java/code/src/com/suse/manager/saltboot/test/PXEEventTest.java

```diff
@@ -235,15 +300,289 @@ public void testParseNoSaltDevice() {
 
         assertTrue(parsed.isPresent());
         parsed.ifPresent(e -> {
-            assertEquals("minion.local", e.getMinionId());
-            assertEquals("groupPrefix", e.getSaltbootGroup());
+            assertEquals(MINION_ID, e.getMinionId());
+            assertEquals(SALTBOOT_GROUP, e.getSaltbootGroup());
             assertEquals("root=/dev/sda1", e.getRoot());
             assertEquals(Optional.empty(), e.getSaltDevice());
             assertEquals(Optional.of("custom=option"), e.getKernelParameters());
-            assertEquals("POS_Image_JeOS7-7.0.0-1", e.getBootImage());
+            assertEquals(BOOT_IMAGE, e.getBootImage());
             // Localhost device should be parsed out
             assertEquals(1, e.getHwAddresses().size());
             assertEquals(e.getHwAddresses().get(0), "00:11:22:33:44:55");
         });
     }
+
+    /**
+     * Tests default PXE event handling
+     *
+     * Processing the event, when repart or redeploy flags are not set and when saltboot-formula is minion formula
+     * @throws Exception
+     */
+    @Test
+    public void testPXEEventAction() throws Exception {
+        MinionServer minion = MinionServerFactoryTest.createTestMinionServer(user);
+
+        Event event = mock(Event.class);
+        Map<String, Object> data = createTestData(minion.getMinionId(), SALTBOOT_GROUP, "root=/dev/sda1",
+                "/dev/sda1", BOOT_IMAGE, "custom=option", true);
+        context().checking(new Expectations() {{
+            allowing(event).getTag();
+            will(returnValue("suse/manager/pxe_update"));
+            allowing(event).getData();
+            will(returnValue(data));
+        }});
+
+        Optional<PXEEvent> pxeevent = PXEEvent.parse(event);
+        assertTrue(pxeevent.isPresent());
+
+        PXEEventMessage pxemsg = new PXEEventMessage(pxeevent.get());
+        PXEEventMessageAction action = new PXEEventMessageAction();
+
+        // Add pillar data without any redeployment flag
+        Set<Pillar> pillars = new HashSet<>();
+        pillars.add(new Pillar(FORMULA, createSaltbootTestData(null, null), minion));
+        minion.setPillars(pillars);
+        TestUtils.saveAndFlush(minion);
+        minion = reload(minion);
```

Comment:
Here and in the other usages below, `saveAndReload()` is meant to both :slightly_smiling_face: 
```suggestion
        minion = TestUtils.saveAndReload(minion);
```